File: assets/codes/leetcode/src/main/java/leetcode/L_49.go
package sample

import "sort"

func groupAnagrams(strs []string) [][]string {
	result := make(map[string][]string)
	for _, s := range strs {
		var data []byte = []byte(s)
		sort.Slice(data, func(i, j int) bool {
			return data[i] > data[j]
		})
		str := string(data)
		if v, e := result[str]; e {
			v = append(v, s)
			result[str] = v
		} else {
			result[str] = []string{s}
		}
	}
	var res = [][]string{}
	for _, strings := range result {
		res = append(res, strings)
	}
	return res
}

File: assets/codes/leetcode/src/main/java/leetcode/L_49.java
package leetcode;

import java.util.*;
import java.util.stream.Collectors;

public class L_49 {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> result = new HashMap<>();

        for (String str : strs) {
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String key = new String(chars);
            List<String> value = result.get(key);
            if (value == null){
                value = new ArrayList<>();
                value.add(str);
                result.put(key,value);
            } else {
                value.add(str);
            }
        }
        return new ArrayList<>(result.values());
    }

    public static void main(String[] args) {
        L_49 l49 = new L_49();
        {
            System.out.println(l49.groupAnagrams(new String[]{"eat", "tea", "tan", "ate", "nat", "bat"}));
        }
        {
            System.out.println(l49.groupAnagrams(new String[]{""}));
        }
        {
            System.out.println(l49.groupAnagrams(new String[]{"a"}));
        }
    }
}
